### ISSUE TITLE
Provide a way to check two contracts share storage layout

### ISSUE THREAD

**Issue by frangio** [zeppelinos/zos-lib#63]
One of the prerequisites for an upgrade to be correct is that the underlying Solidity implementations have the same storage layout. This means that the contracts have the same state variables in the same places, and if there are new state variables in the upgrade they shouldn't conflict with the storage of the state variables in the previous implementation.

We should provide a script to check this. I've begun working on it but I'm blocked by some things about the inheritance graph linearization that I'm waiting to discuss with a few people.

**Comment by maraoz** [comment]
@frangio moved this out of MVP, unless it's close to being done.

**Comment by spalladino** [comment]
That issue seems to be a bit on hold. We may want to resume the discussion with the Solidity maintainers to make a case for that feature to be accepted.

**Comment by maraoz** [comment]
@frangio you should have said ["I accidentally killed it"](https://github.com/paritytech/parity/issues/6995#issue-271523889)

**Comment by frangio** [comment]
We've found that the AST includes the linearization of the inheritance graph. This is info that we can already use for this check. In fact with the storage layout of each separate contract plus the linearization we have nearly everything we need.

**Comment by spalladino** [comment]
Closing in favour of https://github.com/zeppelinos/zos/issues/37
